crates/iota-storage/src/bin/http_kv_tool.rs

```diff
@@ -15,118 +15,173 @@ use iota_types::{
+                        } else {
+                            let fx = kv.multi_get_fx_by_tx_digest(&digests).await.unwrap();
+                            for (digest, fx) in digests.iter().zip(fx.iter()) {
+                                println!("fetched fx: {:?} {:?}", digest, fx);
+                            }
+                        }
+                    }
+
+                    ItemType::CheckpointContents => {
+                        let ckpts = kv.multi_get_checkpoints(&[], &seqs, &[]).await.unwrap();
+
+                        for (seq, ckpt) in seqs.iter().zip(ckpts.1.iter()) {
+                            // populate digest before printing
+                            ckpt.as_ref().map(|c| c.digest());
+                            println!("fetched ckpt contents: {:?} {:?}", seq, ckpt);
+                        }
+                    }
+
+                    ItemType::CheckpointSummary => {
+                        let digests: Vec<_> = digest
+                            .into_iter()
+                            .map(|s| {
+                                CheckpointDigest::from_str(&s).expect("invalid checkpoint digest")
+                            })
+                            .collect();
+
+                        let ckpts = kv
+                            .multi_get_checkpoints(&seqs, &[], &digests)
+                            .await
+                            .unwrap();
+
+                        for (seq, ckpt) in seqs.iter().zip(ckpts.0.iter()) {
+                            // populate digest before printing
+                            ckpt.as_ref().map(|c| c.digest());
+                            println!("fetched ckpt summary: {:?} {:?}", seq, ckpt);
+                        }
+                        for (digest, ckpt) in digests.iter().zip(ckpts.2.iter()) {
+                            // populate digest before printing
+                            ckpt.as_ref().map(|c| c.digest());
+                            println!("fetched ckpt summary: {:?} {:?}", digest, ckpt);
+                        }
+                    }
+
+                    ItemType::Object => {
+                        let object_id = ObjectID::from_str(&digest[0]).expect("invalid object id");
+                        let object = kv.get_object(object_id, seqs[0].into()).await.unwrap();
+                        println!("fetched object {:?}", object);
+                    }
+
+                    _ => {
+                        println!(
+                            "Invalid key type: {}. Must be one of 'tx', 'fx', or 'ev'.",
+                            type_
+                        );
+                        std::process::exit(1);
+                    }
                 }
+                Ok(())
             }
-        }
+            Command::DecodeKey { url } => {
+                // url may look like
+                // https://transactions.iota.io/mainnet/jlkqmZbVuunngIyy2vjBOJSETrM56EH_kIc5wuLvDydN_x0GAAAAAA/ob
```

Comment:
IIRC we don't have `https://transactions.iota.io/`?

Reply:
```diff
-// https://transactions.iota.io/mainnet/jlkqmZbVuunngIyy2vjBOJSETrM56EH_kIc5wuLvDydN_x0GAAAAAA/ob
+// https://{transaction_kv_store_base_url}/jlkqmZbVuunngIyy2vjBOJSETrM56EH_kIc5wuLvDydN_x0GAAAAAA/ob
```